### PR TITLE
Fix audio handling for 48000kHz and add corresponding unit tests

### PR DESCRIPTION
**Pull request by kazukinakai**
Hi @henryruhs,

I have fixed the **48000 kHz bug** I heard from you via Discord DM.  
Please let me know if I am sending pull requests the wrong way.

### Description
This pull request fixes an issue with audio handling when using a sample rate of **48000kHz** in the `ffmpeg.py` file. The changes ensure that the audio sample rate is correctly set during processing, preventing potential errors and improving compatibility with files using this sample rate.

Additionally, a new unit test (`test_audio_48000kHz.py`) has been added to verify the fix and ensure that audio is processed correctly when using **48000kHz**.

### Changes
- **Modified `ffmpeg.py`** to correctly handle audio with a 48000kHz sample rate.
- **Added `test_audio_48000kHz.py`** for unit testing the fix.

### Testing
- Verified the functionality locally by running the unit tests using `unittest`.
- All tests passed successfully, confirming that the bug is resolved.

### Impact
This fix should improve the reliability of audio processing for files using the **48000kHz** sample rate.  
It does not introduce any breaking changes.

### CODE REVIEW

File: facefusion/ffmpeg.py [hunk 1]
```diff
@@ -146,7 +146,20 @@ def restore_audio(target_path : str, output_path : str, output_video_fps : Fps)
 	if isinstance(trim_frame_end, int):
 		end_time = trim_frame_end / output_video_fps
 		commands.extend([ '-to', str(end_time) ])
-	commands.extend([ '-i', target_path, '-c:v', 'copy', '-c:a', state_manager.get_item('output_audio_encoder'), '-map', '0:v:0', '-map', '1:a:0', '-shortest', '-y', output_path ])
+
+	audio_encoder = state_manager.get_item('output_audio_encoder') or 'aac'
+
+	commands.extend([
```
Comment (henryruhs) [inline facefusion/ffmpeg.py]: please keep it on one line like the original

File: tests/test_audio_48000kHz.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import unittest
+from unittest.mock import patch, MagicMock
+from facefusion.ffmpeg import restore_audio
+
+class TestRestoreAudio(unittest.TestCase):
```
Comment (henryruhs) [inline tests/test_audio_48000kHz.py:5]: we are not using classes in this project - just use methods

File: tests/test_audio_48000kHz.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    @patch('facefusion.ffmpeg.state_manager')
+    @patch('facefusion.ffmpeg.get_temp_file_path')
+    @patch('facefusion.ffmpeg.run_ffmpeg')
+    def test_restore_audio(self, mock_run_ffmpeg, mock_get_temp_file_path, mock_state_manager):
```
Comment (henryruhs) [inline tests/test_audio_48000kHz.py:10]: I would actual say, just write a cli based integration test - this looks over the top with all the mocks

File: facefusion/ffmpeg.py [hunk 4]
```diff
@@ -146,7 +146,20 @@ def restore_audio(target_path : str, output_path : str, output_video_fps : Fps)
...
+        '-i', target_path,
+        '-c:v', 'copy',
+        '-c:a', audio_encoder,
+        '-ar', '48000',  # Sample rate set to 48 kHz
```
Comment (henryruhs) [inline facefusion/ffmpeg.py]: Remove comments

**Review by henryruhs — CHANGES_REQUESTED** [review]

File: facefusion/ffmpeg.py (hunk 1 above)
Comment (kazukinakai) [reply facefusion/ffmpeg.py]: I fix the keep one line code like original.

File: tests/test_audio_48000kHz.py (hunk 2 above)
Comment (kazukinakai) [reply tests/test_audio_48000kHz.py]: I fixed classes to methods

**Review by kazukinakai — COMMENTED** [review]
Hi @henryruhs 
Thank you for your kind review.
I noticed the description here when I revised it based on the Feedbacks from you. lol

File: facefusion/ffmpeg.py (hunk 4 above)
Comment (kazukinakai) [reply facefusion/ffmpeg.py]: I delete comments

**Comment by kazukinakai** [comment]
Thank you for your kind review.
I noticed the description here when I revised it based on the Feedbacks from you. lol

**Review by kazukinakai — COMMENTED** [review]
checked

**Comment by henryruhs** [comment]
I closed the pull request. In the next branch, I added unit tests for restore_audio(), and it appears to work with a 48kHz file.

**Pull request closed without merging**